infra/lambda_functions/websocket/disconnect/app.js

```diff
@@ -44,43 +49,113 @@ exports.handler = async (event) => {
   };
 
   console.log("openDiagramsScanParams:", openDiagramsScanParams);
-
+  let connectionsToRemove;
   try {
-    const connectionsToRemove = await ddb
-      .scan(openDiagramsScanParams)
-      .promise();
+    connectionsToRemove = await ddb.scan(openDiagramsScanParams).promise();
     console.log("connectionsToRemove: ", connectionsToRemove);
+  } catch (e) {
+    console.log("Error when deleting connection: ", e);
+    // return {
+    //   statusCode: 500,
+    //   body: "Failed to disconnect: " + JSON.stringify(e),
+    // };
+  }
 
-    const connectionToRemove = connectionsToRemove.Items[0];
-    if (!connectionToRemove) {
-      console.log("No connection to remove from open diagrams table");
-    } else {
-      console.log("Attempting to remove connection:", connectionToRemove);
-      const deleteParamsOpenDiagramConnection = {
-        TableName: OPEN_DIAGRAMS_TABLE_NAME,
-        Key: {
-          connectionId: connectionToRemove.connectionId,
-          diagramId: connectionToRemove.diagramId,
-        },
-      };
-      try {
-        console.log(
-          "deleteParamsConnection:",
-          deleteParamsOpenDiagramConnection
-        );
-        await ddb.delete(deleteParamsOpenDiagramConnection).promise();
-        console.log("Successfully deleted connection:", connectionToRemove);
-      } catch (e) {
-        console.log("Error when deleting connection: ", e);
-      }
-    }
+  const connectionToRemove = connectionsToRemove.Items[0];
```

Comment:
shall we not check the length before trying to pull out the 0th element? this can error otherwise.